From c5d3fdce483738f0be6aa93c4ed139bbb6fc4240 Mon Sep 17 00:00:00 2001
From: Emilia Zapata <zapata.contact@gmail.com>
Date: Sun, 21 Aug 2022 15:53:03 +1200
Subject: [PATCH] feat(git_status): track staged/unstaged deletes

---
 src/modules/git_status.rs | 15 ++++++++++-----
 1 file changed, 10 insertions(+), 5 deletions(-)

diff --git a/src/modules/git_status.rs b/src/modules/git_status.rs
index aa3f0387882..9e454628699 100644
--- a/src/modules/git_status.rs
+++ b/src/modules/git_status.rs
@@ -264,13 +264,13 @@ struct RepoStatus {

 impl RepoStatus {
     fn is_deleted(short_status: &str) -> bool {
-        // is_wt_deleted || is_index_deleted
-        short_status.contains('D')
+        // is_index_deleted
+        short_status.starts_with('D')
     }

     fn is_modified(short_status: &str) -> bool {
-        // is_wt_modified || is_wt_added
-        short_status.ends_with('M') || short_status.ends_with('A')
+        // is_wt_modified || is_wt_added || is_wt_deleted
+        short_status.ends_with('M') || short_status.ends_with('A') || short_status.ends_with('D')
     }

     fn is_staged(short_status: &str) -> bool {
